src/commands/startStreamingLogs.ts

```diff
@@ -5,27 +5,40 @@
 
 import * as vscode from 'vscode';
 import * as appservice from 'vscode-azureappservice';
+import { LogStreamTreeItem } from '../explorer/LogStreamTreeItem';
 import { SiteTreeItem } from "../explorer/SiteTreeItem";
 import { WebAppTreeItem } from '../explorer/WebAppTreeItem';
 import { ext } from '../extensionVariables';
 import { enableFileLogging } from './enableFileLogging';
 
-export async function startStreamingLogs(node?: SiteTreeItem): Promise<void> {
+export async function startStreamingLogs(node?: SiteTreeItem | LogStreamTreeItem): Promise<void> {
+    let logStreamTreeItem: LogStreamTreeItem | undefined;
+    let webAppTreeItem: WebAppTreeItem;
+
     if (!node) {
-        node = <WebAppTreeItem>await ext.tree.showTreeItemPicker(WebAppTreeItem.contextValue);
+        webAppTreeItem = <WebAppTreeItem>await ext.tree.showTreeItemPicker(WebAppTreeItem.contextValue);
+    } else if (node.contextValue === LogStreamTreeItem.contextValue) {
+        logStreamTreeItem = <LogStreamTreeItem>node;
+        // tslint:disable-next-line:no-non-null-assertion
+        webAppTreeItem = <WebAppTreeItem>node.parent!.parent;
```

Comment:
If you define the type of the parent like this: https://github.com/Microsoft/vscode-azureappservice/blob/master/src/explorer/CosmosDBTreeItem.ts#L19

Than you won't have to do the non-null-assertion

Reply:
I'm having some issues with this.. :|